modules/ztp-image-based-upgrade-prep-oadp.adoc

```diff
@@ -0,0 +1,136 @@
+= Creating OADP resources for the image-based upgrade with GitOps ZTP
+
+You need to prepare your OADP resources to restore your application after an upgrade.
+
+.Prerequisites
+
+* Provision one or more managed clusters with GitOps ZTP.
+* Log in as a user with `cluster-admin` privileges.
+* Generate a seed image from a compatible seed cluster.
+* Create a separate partition on the target cluster for the container images that is shared between stateroots. For more information about, see _Additional resources_.
+* Deploy a version of {lcao} that is compatible with the version used with the seed image.
+* Install the OADP Operator, the `DataProtectionApplication` CR and its secret on the target cluster.
+* Create an S3-compatible storage solution and a ready-to-use bucket with proper credentials configured. For more information, see Additional resources.
+
+.Procedure
+
+. Ensure that your Git repository that you use with the ArgoCD policies application contains the following directory structure:
++
+--
+[source,terminal]
+----
+├── source-crs/
+│   ├── ibu/
+│   │    ├── ImageBasedUpgrade.yaml
+│   │    ├── PlatformBackupRestore.yaml
+│   │    ├── PlatformBackupRestoreLvms.yaml
+├── ...
+├── ibu-upgrade-ranGen.yaml
+├── kustomization.yaml
+----
+
+[IMPORTANT]
+====
+The `kustomization.yaml` file must be located in the same directory structure as shown above to reference the `PlatformBackupRestore.yaml` manifest.
+====
+
+The `source-crs/ibu/PlatformBackupRestore.yaml` file is provided in the ZTP container image.
+
+.PlatformBackupRestore.yaml
+include::snippets/ibu-PlatformBackupRestore.adoc[]
+
+If you use {lvms} to create persistent volumes, you can use the `source-crs/ibu/PlatformBackupRestoreLvms.yaml` provided in the ZTP container image to back up your {lvms} resources.
+
+.PlatformBackupRestoreLvms.yaml
+include::snippets/ibu-PlatformBackupRestoreLvms.adoc[]
+--
+
+. (Optional) If you need to restore applications over the upgrade, create the OADP `Backup` and `Restore` CRs for your application in the `openshift-adp` namespace.
+
+.. Create the OADP CRs for cluster-scoped application artifacts in the `openshift-adp` namespace.
++
+.Example OADP CRs for cluster-scoped application artifacts for LSO and {LVMS}
+include::snippets/ibu-ApplicationClusterScopedBackupRestore.adoc[]
+
+.. Create the OADP CRs for your namespace-scoped application artifacts in the `source-crs/custom-crs` directory.
++
+--
+.Example OADP CRs namespace-scoped application artifacts when LSO is used
+include::snippets/ibu-ApplicationBackupRestoreLso.adoc[]
+
+.Example OADP CRs namespace-scoped application artifacts when {lvms} is used
+include::snippets/ibu-ApplicationBackupRestoreLvms.adoc[]
+
+[IMPORTANT]
+====
+The same version of the applications must function on both the current and the target release of {product-title}.
+====
+--
+
+. Create the `oadp-cm` `ConfigMap` object through the `oadp-cm-policy` in a new `PolicyGenTemplate` called `ibu-upgrade-ranGen.yaml`.
```

Comment:
Make sure to use nouns after object references and file names: https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc#object-references